scripts/performance/benchmark_utils.py

```diff
@@ -0,0 +1,119 @@
+KB = 1024
+SIZE_SUFFIX = {
+    'kb': 1024,
+    'mb': 1024 ** 2,
+    'gb': 1024 ** 3,
+    'tb': 1024 ** 4,
+    'kib': 1024,
+    'mib': 1024 ** 2,
+    'gib': 1024 ** 3,
+    'tib': 1024 ** 4,
+}
+
+
+def human_readable_to_bytes(value):
+    """Converts a human readable size to bytes.
+    :param value: A string such as "10MB".  If a suffix is not included,
+        then the value is assumed to be an integer representing the size
+        in bytes.
+    :returns: The converted value in bytes as an integer
+    """
+    value = value.lower()
+    if value[-2:] == 'ib':
+        # Assume IEC suffix.
+        suffix = value[-3:].lower()
+    else:
+        suffix = value[-2:].lower()
+    has_size_identifier = (
+        len(value) >= 2 and suffix in SIZE_SUFFIX)
+    if not has_size_identifier:
+        try:
+            return int(value)
+        except ValueError:
+            raise ValueError("Invalid size value: %s" % value)
+    else:
+        multiplier = SIZE_SUFFIX[suffix]
+        return int(value[:-len(suffix)]) * multiplier
+
+
+def summarize(script, directory, filename='summary.txt'):
+    summarize_args = [script]
+    for f in os.listdir(directory):
+        path = os.path.join(directory, f)
+        if os.path.isfile(path):
+            summarize_args.append(path)
+
+    with open(os.path.join(directory, filename), 'wb') as f:
+        subprocess.check_call(summarize_args, stdout=f)
+
+
+def _get_s3transfer_performance_script(script_name):
+    """Retrieves an s3transfer performance script if available."""
+    s3transfer_directory = os.path.dirname(s3transfer.__file__)
+    s3transfer_directory = os.path.dirname(s3transfer_directory)
+    scripts_directory = 'scripts/performance'
+    scripts_directory = os.path.join(s3transfer_directory, scripts_directory)
+    script = os.path.join(scripts_directory, script_name)
+
+    if os.path.isfile(script):
+        return script
+    else:
+        return None
+
+
+def get_benchmark_script():
+    return _get_s3transfer_performance_script('benchmark')
+
+
+def get_summarize_script():
+    return _get_s3transfer_performance_script('summarize')
+
+
+def backup(source, recursive):
+    if source[:5] == 's3://':
+        parts = source.split('/')
+        parts.insert(3, str(uuid.uuid4()))
+        backup_path = '/'.join(parts)
+    else:
+        name = os.path.split(source)[-1]
+        temp_dir = tempfile.mkdtemp()
+        backup_path = os.path.join(temp_dir, name)
+
+    copy(source, backup_path, recursive)
+    return backup_path
+
+
+def copy(source, destination, recursive):
+    if 's3://' in [source[:5], destination[:5]]:
+        cp_args = ['aws', 's3', 'cp', source, destination, '--quiet']
+        if recursive:
+            cp_args.append('--recursive')
+        subprocess.check_call(cp_args)
+        return
+
+    if recursive:
+        shutil.copytree(source, destination)
+    else:
+        shutil.copy(source, destination)
+
+
+def clean(destination, recursive):
```

Comment:
I think we need to be careful about this.  Especially in the `--recursive` case, if I had specified my `~/` directory as the dest and I was downloading objects from S3 that were broken down by prefix:

```
s3://bucket/foo
s3://bucket/bar
```

Then this script (with a source of `s3://bucket/`) would download to `~/foo` and `~/bar` if I was testing `s3->local`, but as part of cleanup it would `rm -rf` my entire home directory.

I think we should try to be more conservative in how my cleanup.  The ideal result is we only remove what we downloaded/uploaded, but I realize we would need to track that information somehow.  Is there a safer way we can clean up?